src/rgw/store/sfs/sfs_bucket.cc

```diff
@@ -43,8 +43,7 @@ int SFStore::get_bucket(const DoutPrefixProvider *dpp, User *u,
     return -ENOENT;
   }
   auto bucketref = it->second;
-  RGWBucketInfo info;
-  auto bucket = make_unique<SFSBucket>(this, bucketref, bucketref->to_rgw_bucket_info(info));
+  auto bucket = make_unique<SFSBucket>(this, bucketref, bucketref->to_rgw_bucket_info());
```

Comment:
any particular reason why we're doing a copy here?

Reply:
Can you explain better what you mean?
This is just a style patch not directly related to this PR; I used this PR to do this and not opened a dedicated one because it is (is it :smirk: ?)  reasonably safe.

Reply:
I think this is related to a discussion we had in a previous PR.
In this particular case, I think the compiler will be doing copy elision and moving the structure (returning its local variable), so no copy should be involved as this is used in a constructor as destination memory is not initialised before the call.

Reply:
Yes I remember that.
An ever more efficient operation here would be to remove the 3rd param from the `SFSBucket` constructor and call `to_rgw_bucket_info` over the bucket object (so no copy of info would exist at all).
But considering that we are talking about a `create` bucket operation I don't think we have a special need to super-optimize this call.

Reply:
yes, removing the 3rd parameter and doing it in the constructor sounds good to me, but I would not consider it a blocker for approving this PR.
Anyway.. it is something that we do every time we construct the `SFSBucket` when we pass the `sfs::BucketRef`, right? 
Makes sense doing it in that particular constructor.

Reply:
yes, it is something we do at every `SFSBucket` construction.
ok, I will modify that (removing 3rd param and do it insisde ctor), better do it now.

Reply:
mhmmh, no, this is not safe as I thought, doing this change in the context of this PR is potentially less safe than doing it in a dedicated one.
The problem here is that we have to rely on transformations from `rgw::sal::sfs::Bucket` to get a `RGWBucketInfo`.
I think something on `rgw::sal::sfs::Bucket` is missing; IMO the best choice for a future refactoring is to consider the idea to drop entirely `rgw::sal::sfs::Bucket` and rely only on `SFSBucket`.

Reply:
How do you suggest we track bucket-specific state with just `SFSBucket`?

Reply:
I don't know; I was just wondering if could be an idea for the future try to simplify this aspect.
But perhaps there are constraint that are preventing this.
Having just one subclass for sfs bucket is something desirable because it avoid completely to transfer information from one type to another.
But if we cannot have this, "coraggio" (courage :).

Reply:
I think we can close this thread, if anyone thinks that there is a better approach to model bucket objects in `sfs` is always welcome to propose an idea.
From my side, if I will have time, I will try to hack something.